application.py

```diff
@@ -77,29 +77,47 @@ def allowed_file(filename):
 
 @application.route('/api/image', methods=['POST'])
 def generate_image():
-    # check if the post request has the file part
-    if 'file' not in request.files:
-        abort(400, "No 'file' in the request.")
-    file = request.files['file']
-    # if user does not select file, browser also
-    # submit an empty part without filename
-    if file.filename == '':
-        abort(400, "The filename  in the request was empty.")
-
-    if not allowed_file(file.filename):
-        abort(400, "Bad file extension, only '.wrl' filetypes are allowed")
-
-    if file and allowed_file(file.filename):
-        # Save the uploaded file to a temporary file and then pass
-        # that on to the matlab image generation code
-        with NamedTemporaryFile() as f:
-            file.save(f.name)
-            image_fname = call_matlab_image(f.name)
-            return send_file(image_fname, mimetype='image/png')
+    # the file collection is lost when requests go through the API Gateway http proxy
+    file = request.files.get('file') or request.environ.get('wsgi.input')
+    if not file:
+        abort(411, f"file or wsgi.input argument is required\n{pprint.pformat(('REQUEST', request.environ))}")
+
+    # get the file path if possible
+    # it works for local flask mode and for direct access to EB instance
+    # it doesn't work for API Gateway http proxy in front of EB instance
+    file_path = getattr(file, 'filename', None)
+    if file_path and not allowed_file(file_path):
+        abort(400, f"Bad file extension, only '.wrl' filetypes are allowed in '{file_path}'")
+
+    # Save the uploaded file to a temporary file and then pass
+    # that on to the matlab image generation code
+    bufsize = 16384
+    with NamedTemporaryFile() as f:
+        # read the post data stream using a reasonable buffer size
+        # the save method isn't available on the wsgi.input data stream
+        data = file.read(bufsize)
+        while data:
+            f.write(data)
+            data = file.read(bufsize)
+        
+        image_fname = call_matlab_image(f.name)
+        return send_file(image_fname, mimetype='image/png')
 
     # If we made it here, there was an unknown error
     abort(400, "Unknown error handling the uploaded image")
 
 
+@application.errorhandler(Exception)
+def handle_error(error):
```

Comment:
this provides a lot of information in the response when something fails in the back end.